src/x509_str.c

```diff
@@ -53,7 +65,12 @@ WOLFSSL_X509_STORE_CTX* wolfSSL_X509_STORE_CTX_new_ex(void* heap)
         XMEMSET(ctx, 0, sizeof(WOLFSSL_X509_STORE_CTX));
         ctx->heap = heap;
 #ifdef OPENSSL_EXTRA
-        if (wolfSSL_X509_STORE_CTX_init(ctx, NULL, NULL, NULL) !=
+        if ((ctx->owned = wolfSSL_sk_X509_new_null()) == NULL) {
+            XFREE(ctx, heap, DYNAMIC_TYPE_X509_CTX);
+            ctx = NULL;
+        }
+        if (ctx != NULL &&
+            wolfSSL_X509_STORE_CTX_init(ctx, NULL, NULL, NULL) !=
                 WOLFSSL_SUCCESS) {
             XFREE(ctx, heap, DYNAMIC_TYPE_X509_CTX);
             ctx = NULL;
```

Comment:
I think its safer to call `wolfSSL_X509_STORE_CTX_free` on error here. The pain point would be if `wolfSSL_X509_STORE_CTX_init` allocates something then we don't free it here.

Reply:
I agree, good catch.